packages/contracts/contracts/libs/v0.8.x/ERC721_PackedHashSeedV1.sol

```diff
@@ -0,0 +1,592 @@
+// SPDX-License-Identifier: MIT
+// OpenZeppelin Contracts (last updated v5.0.0) (token/ERC721/ERC721.sol)
+
+pragma solidity ^0.8.20;
+
+import "@openzeppelin-5.0/contracts/token/ERC721/IERC721.sol";
+import "@openzeppelin-5.0/contracts/token/ERC721/IERC721Receiver.sol";
+import "@openzeppelin-5.0/contracts/token/ERC721/extensions/IERC721Metadata.sol";
+import "@openzeppelin-5.0/contracts/utils/Address.sol";
+import "@openzeppelin-5.0/contracts/utils/Context.sol";
+import "@openzeppelin-5.0/contracts/utils/Strings.sol";
+import "@openzeppelin-5.0/contracts/utils/introspection/ERC165.sol";
+import "@openzeppelin-5.0/contracts/interfaces/draft-IERC6093.sol";
+
+/**
+ * @dev Forked version of the OpenZeppelin v5.0.0 ERC721 contract. Updated
+ * with an initialize function to ensure EIP 1167 compatibility. Utilizes
+ * a struct to pack owner and hash seed into a single storage slot.
+ * ---------------------
+ * @dev Implementation of https://eips.ethereum.org/EIPS/eip-721[ERC721] Non-Fungible Token Standard, including
+ * the Metadata extension, but not including the Enumerable extension, which is available separately as
+ * {ERC721Enumerable}.
+ */
+contract ERC721_PackedHashSeedV1 is
+    Context,
+    ERC165,
+    IERC721,
+    IERC721Metadata,
+    IERC721Errors
+{
+    using Address for address;
+    using Strings for uint256;
+
+    // Token name
+    string private _name;
+
+    // Token symbol
+    string private _symbol;
+
+    /// ensure initialization can only be performed once
+    bool private initialized;
+
+    /// struct to pack a token owner and hash seed into same storage slot
+    struct OwnerAndHashSeed {
+        // 20 bytes for address of token's owner
+        address owner;
+        // remaining 12 bytes allocated to token hash seed
+        bytes12 hashSeed;
+    }
+
+    /// mapping of token ID to OwnerAndHashSeed
+    /// @dev visibility internal so inheriting contracts can access
+    mapping(uint256 => OwnerAndHashSeed) internal _ownersAndHashSeeds;
+
+    // Mapping owner address to token count
+    mapping(address => uint256) private _balances;
+
+    // Mapping from token ID to approved address
+    mapping(uint256 => address) private _tokenApprovals;
+
+    // Mapping from owner to operator approvals
+    mapping(address => mapping(address => bool)) private _operatorApprovals;
```

Comment:
nit: can you used named variables in the mappings to stay consistent with OpenZeppelin implementation?

Reply:
Good call! Updated 🙏